zcash_client_sqlite/src/wallet/init/migrations/receiving_key_scopes.rs

```diff
@@ -165,21 +168,29 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                     // be mined under ZIP 212 enforcement rules, so we default to `On`
                     Zip212Enforcement::On
                 },
-                |h| sapling_zip212_enforcement(&self.params, h),
+                |h| zip212_enforcement(&self.params, h),
             );
 
             let ufvk_str: String = row.get(5)?;
-            let ufvk = UnifiedFullViewingKey::decode(&self.params, &ufvk_str)
-                .expect("Stored UFVKs must be valid");
-            let dfvk = ufvk
-                .sapling()
-                .expect("UFVK must have a Sapling component to have received Sapling notes");
+            let ufvk = UnifiedFullViewingKey::decode(&self.params, &ufvk_str).map_err(|e| {
+                WalletMigrationError::CorruptedData(format!("Stored UFVK was invalid: {:?}", e))
+            })?;
+
+            let dfvk = ufvk.sapling().ok_or_else(|| {
+                WalletMigrationError::CorruptedData(
+                    "UFVK must have a Sapling component to have received Sapling notes.".to_owned(),
+                )
+            })?;
 
             // We previously set the default to external scope, so we now verify whether the output
             // is decryptable using the intenally-scoped IVK and, if so, mark it as such.
             if let Some(tx_data) = tx_data_opt {
-                let tx = Transaction::read(&tx_data[..], BranchId::Canopy)
-                    .expect("Transaction must be valid");
+                let tx = Transaction::read(&tx_data[..], BranchId::Canopy).map_err(|e| {
+                    WalletMigrationError::CorruptedData(format!(
+                        "Unable to parse raw transaction: {:?}",
+                        e
+                    ))
+                })?;
                 let output = tx
                     .sapling_bundle()
                     .and_then(|b| b.shielded_outputs().get(output_index))
```

Comment:
Should the panic just below also be a `WalletMigrationError::CorruptedData` return?

Reply:
Nit: we sometimes use `"string literal".to_owned()`, and sometimes `"string literal".to_string()`. They [do the same thing](https://users.rust-lang.org/t/current-guidance-on-str-to-string-to-owned-vs-to-string/75754), but let's be consistent and use `to_owned()`.

Reply:
We are so totally inconsistent about this that fixing it should be done outside of this PR; I find 70 different occurrences of `.to_string` all across the codebase.